### ISSUE TITLE
Implement rust-analyzer's colorDiagnosticOutput 

### ISSUE THREAD

**Issue by sclu1034** [kak-lsp/kak-lsp#665]
rust-analyzer can now provide full-color diagnostics (presumably ANSI codes, as returned by Cargo) when configured to do so.
Similar to parsing Markdown into faces in the info box, these could be parsed as well.

[low priority, probably candidate for 'good first issue']

**Comment by krobelus** [comment]
good idea, this can apply to the info box and other places where we show diagnostics (like the buffer created by `lsp-diagnostics`)
